src/main/java/yablunovskyi/bookstore/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,113 @@
+package yablunovskyi.bookstore.service.impl;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import yablunovskyi.bookstore.dto.shoppingcart.CartResponseDto;
+import yablunovskyi.bookstore.dto.shoppingcart.CreateCartItemRequestDto;
+import yablunovskyi.bookstore.dto.shoppingcart.UpdateCartItemRequestDto;
+import yablunovskyi.bookstore.exception.EntityNotFoundException;
+import yablunovskyi.bookstore.mapper.CartItemMapper;
+import yablunovskyi.bookstore.mapper.ShoppingCartMapper;
+import yablunovskyi.bookstore.model.CartItem;
+import yablunovskyi.bookstore.model.ShoppingCart;
+import yablunovskyi.bookstore.model.User;
+import yablunovskyi.bookstore.repository.BookRepository;
+import yablunovskyi.bookstore.repository.CartItemRepository;
+import yablunovskyi.bookstore.repository.ShoppingCartRepository;
+import yablunovskyi.bookstore.repository.UserRepository;
+import yablunovskyi.bookstore.service.ShoppingCartService;
+
+@Service
+@RequiredArgsConstructor
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final BookRepository bookRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final CartItemRepository cartItemRepository;
+    private final UserRepository userRepository;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final CartItemMapper cartItemMapper;
+    
+    @Override
+    @Transactional
+    public CartResponseDto addCartItem(
+            Authentication authentication, CreateCartItemRequestDto requestDto) {
+        checkIfBookExists(requestDto.bookId());
```

Comment:
Yes, because according to my logic if book with request id doesn't exist I will create an item with a unexisted book. So to prevent it I need this check.